stock_manager/controllers/abstract.py

```
@@ -167,9 +178,6 @@ def __init__(self, file_name: str, app: 'App'):
 	def to_page(self) -> None:
 		"""Navigate to this scanner page and start the video feed."""
 		
-		self.app.current_page = self.PAGE_NAME
-		self.app.screens.setCurrentIndex(self.PAGE_NAME.value.PAGE_INDEX)
-		
 		try:
 			self.start_video()
 		except Exception as e:
```

It's bad form to catch generic exceptions.  You should be specific about what exception you want to catch, and only catch that exception. 

If your code throws an error you don't expect, it may still be caught by this block and considered "ok".

---

I understand this and where I can I do catch specific exceptions, however with python packages that I use in this project, it doesn't tell me what errors could arise or even that there could be errors at all. With languages that I have experience with like Java, anything that can throw an error either directly tell you in the documentation or displays in the IDE. It seems that isn't the case with python, and I am unaware of how to find out what errors something can raise. I think my python error handling skills could use some work, I feel my error handling throughout this entire project is pretty basic and lacking.

---

You're right that normally it should be easier to find out what specific errors might get thrown, and well written python should also do this.  If you're having trouble finding an appropriate exception to catch here, the code probably doesn't expect to raise errors (or the library isn't thorough)

At this point I think it's less a problem with your skills and more a problem with whatever dependencies you're using.  In this case I saw we were catching an error from one of our methods, and figured we were working around a known error.

---

Not exactly, I wrapped `.start_video()` in a `try-except` because if an error occurs in that method it would be logged properly. However, something that I cant seem to figure out is whether python has a sort of "error chaining" that Java has. In Java if a method throws an error, even if the error is handled, it is still passed up through all the methods, being logged in multiple places for the same singular issue, that is why I very commonly wrap method calls in error handlers even if there are already error handlers in those methods. But since this method is only being called once and everything in the method is already wrapped in a `try-except` I'm not sure if the `try-except` in `.to_page()` is necessary. If not, I could also look throughout my program and remove any unnecessary error handling that I may have in place.

---

I see where you're coming from.  To the best of my understanding / memory, exception handling is similar between java and python.  In python, if an exception is thrown and not caught, it will print a stack trace and end your program.  If it's caught, you get to choose what to do do next, and the error propagation ends there. 

I would argue that unless you expect the app to throw an error during **normal** operation, you let the error propagate and end the program.  If you do expect it, you handle it appropriately for that context.  Otherwise the app can break silently